src/widgets/3dtexture/lv_3dtexture.c

```diff
@@ -71,6 +71,15 @@ void lv_3dtexture_set_src(lv_obj_t * obj, lv_3dtexture_id_t id)
     tex->id = id;
 }
 
+void lv_3dtexture_set_src_flip(lv_obj_t * obj, bool h_flip, bool v_flip)
+{
+    LV_ASSERT_OBJ(obj, MY_CLASS);
+
+    lv_3dtexture_t * tex = (lv_3dtexture_t *)obj;
+    tex->h_flip = h_flip;
+    tex->v_flip = v_flip;
+}
+
```

Comment:
The important part of this suggestion is `lv_obj_invalidate(obj);` and the `if` is an optimization.

```suggestion
void lv_3dtexture_set_src_flip(lv_obj_t * obj, bool h_flip, bool v_flip)
{
    LV_ASSERT_OBJ(obj, MY_CLASS);

    lv_3dtexture_t * tex = (lv_3dtexture_t *)obj;
    
    if(tex->h_flip == h_flip && tex->v_flip == v_flip) return;
    
    tex->h_flip = h_flip;
    tex->v_flip = v_flip;
    
    lv_obj_invalidate(obj);
}

```